lib/Zonemaster/Backend/Config.pm

```diff
@@ -24,151 +24,167 @@ else {
-    $nb = $cfg->val( 'ZONEMASTER', 'number_of_processes_for_frontend_testing' ) unless ($nb);
+    my $nb = $self->{cfg}->val( 'ZONEMASTER', 'number_of_professes_for_frontend_testing' );
+    $nb = $self->{cfg}->val( 'ZONEMASTER', 'number_of_processes_for_frontend_testing' ) unless ($nb);
     
     return $nb;
 }
 
 sub NumberOfProcessesForBatchTesting {
-    my $cfg = _load_config();
+    my ($self) = @_;
 
-    my $nb = $cfg->val( 'ZONEMASTER', 'number_of_professes_for_batch_testing' );
-    $nb = $cfg->val( 'ZONEMASTER', 'number_of_processes_for_batch_testing' ) unless ($nb);
+    my $nb = $self->{cfg}->val( 'ZONEMASTER', 'number_of_professes_for_batch_testing' );
+    $nb = $self->{cfg}->val( 'ZONEMASTER', 'number_of_processes_for_batch_testing' ) unless ($nb);
     
     return $nb;
 }
 
 sub Maxmind_ISP_DB_File {
-    my $cfg = _load_config();
+    my ($self) = @_;
 
-    return $cfg->val( 'GEOLOCATION', 'maxmind_isp_db_file' );
+    return $self->{cfg}->val( 'GEOLOCATION', 'maxmind_isp_db_file' );
 }
 
 sub Maxmind_City_DB_File {
-    my $cfg = _load_config();
+    my ($self) = @_;
 
-    return $cfg->val( 'GEOLOCATION', 'maxmind_city_db_file' );
+    return $self->{cfg}->val( 'GEOLOCATION', 'maxmind_city_db_file' );
 }
 
 sub force_hash_id_use_in_API_starting_from_id {
-    my $cfg = _load_config();
+    my ($self) = @_;
 
-    my $val = $cfg->val( 'ZONEMASTER', 'force_hash_id_use_in_API_starting_from_id' );
+    my $val = $self->{cfg}->val( 'ZONEMASTER', 'force_hash_id_use_in_API_starting_from_id' );
 
     return ($val)?($val):(0);
 }
 
-sub CustomProfilesPath {
-    my $cfg = _load_config();
+sub ReadProfilesInfo {
+    my ($self) = @_;
+    
+    my $profiles;
+    foreach my $public_profile ($self->{cfg}->Parameters('PUBLIC PROFILES')) {
+        $profiles->{lc($public_profile)}->{type} = 'public';
+        $profiles->{lc($public_profile)}->{profile_file_name} = $self->{cfg}->val('PUBLIC PROFILES', $public_profile);
+    }
 
-    my $value  = $cfg->val( 'ZONEMASTER', 'cutom_profiles_path' );
-    $value  = $cfg->val( 'ZONEMASTER', 'custom_profiles_path' ) unless ($value);
-    return $value;
+    foreach my $private_profile ($self->{cfg}->Parameters('PRIVATE PROFILES')) {
+        $profiles->{lc($private_profile)}->{type} = 'private';
+        $profiles->{lc($private_profile)}->{profile_file_name} = $self->{cfg}->val('PRIVATE PROFILES', $private_profile);
+    }
+    
+    return $profiles;
 }
 
-sub GetCustomConfigParameter {
-	my ($slef, $section, $param_name) = @_;
-	
-    my $cfg = _load_config();
+sub ListPublicProfiles {
+    my ($self) = @_;
+    
+    my $profiles = [];
+    foreach my $public_profile ($self->{cfg}->Parameters('PUBLIC PROFILES')) {
+        $profiles->{lc($public_profile)}->{type} = 'public';
+        $profiles->{lc($public_profile)}->{profile_file_name} = $self->{cfg}->val('PUBLIC PROFILES', $public_profile);
+    }
 
-    return $cfg->val( $section, $param_name );
+    return @$profiles;
```

Comment:
Will this just return an array of public profile names and nothing else?

Reply:
Fix in commit cda24a7 
It will return just the profile names.